### ISSUE TITLE
Value cannot be opened: "BrowserName" when any search is made

### ISSUE THREAD

**Issue by loganran** [spikespaz/search-deflector#10]
I have encountered an error launching Search Deflector.
The error information follows below.

**File:** `std\windows\registry.d`

**Line:** 495

**Message:**
```
Value cannot be opened: "BrowserName": The system cannot find the file specified. (error 2)
```

**Stack Trace:**
```
0x00007FF68F28B63D
0x00007FF68F28B396
0x00007FF68F28424C
0x00007FF68F27106C
0x00007FF68F24EEFA
0x00007FF68F250D48
0x00007FF68F241642
0x00007FF68F241364
0x00007FF68F27727D
0x00007FF68F27712A
0x00007FF68F295BC0
0x00007FFF66A83034 in BaseThreadInitThunk
0x00007FFF671A1431 in RtlUserThreadStart

```

**Comment by spikespaz** [comment]
Do you have any antivirus software besides Windows Defender running? Maybe a 3rd party uninstaller program?

Did you close the console that opened during setup without completing the prompts?
Follow all of the steps in [the README](https://github.com/spikespaz/search-deflector#setup).

I think you may have done something wrong here when setting it up because that registry value should exist unless something stopped it from existing.

**Comment by loganran** [comment]
There is a possibility that Sophos is stopping it. This is on my work
computer so I'll see about stopping Sophos to configure it again. I did get
through all the prompts and hit enter to close.

**Comment by tango409** [comment]
I am also running Sophos Endpoint, but I am not getting anything in the Sophos events.  This error happens immediately when I run the Configure app.

**Comment by spikespaz** [comment]
Root: HKCU; Subkey: "Software\Clients\SearchDeflector"; Flags: uninsdeletekey
Root: HKLM; Subkey: "Software\Clients\SearchDeflector"; Flags: uninsdeletekey
Root: HKLM; SubKey: "Software\RegisteredApplications"; ValueName: "SearchDeflector"; Flags: uninsdeletevalue
Root: HKCR; Subkey: "HKEY_CURRENT_USER\\SearchDeflector"

**Comment by spikespaz** [comment]
So the thing is even though you said that program doesn't appear to be causing the issue, I think it is. It's the only common factor between you too. Maybe just having Sophos installed is preventing the setup from writing to the registry.

The installer clearly thinks that it's finished writing to the registry, otherwise you would be reporting a different issue.

**Comment by spikespaz** [comment]
Please open `regedit` and in the text box on top, navigate to `HKEY_CURRENT_USER\Software\Clients\SearchDeflector`.

Click File > Export, and save the `.reg` file to your desktop. Right click that file, click Edit, and copy/paste the text in a new comment here.

Thanks.

**Comment by tango409** [comment]
There is nothing there.

[HKEY_CURRENT_USER\Software\Clients\SearchDeflector]

**Comment by spikespaz** [comment]
I should have thought of this a while ago, but did you run the installer as an Administrator?

**Comment by loganran** [comment]
I ran it as admin and as regular user. Administrator is what caused the
issue. Regular user mode just said access denied.

**Comment by tango409** [comment]
I've tried Run as Administrator, and just regularly.  Either method will present a UAC prompt.

**Comment by spikespaz** [comment]
@loganran @tango409 Try running the latest release's setup.

https://github.com/spikespaz/search-deflector/releases

**Comment by tango409** [comment]
Release v0.2.0 same results.

**File:** `std\windows\registry.d`

**Line:** 495

**Message:**
```
Value cannot be opened: "": The system cannot find the file specified. (error 2)
```

**Stack Trace:**
```
0x00007FF703390F5D
0x00007FF703390CB6
0x00007FF703389B6C
0x00007FF70337660C
0x00007FF7033299DA
0x00007FF70332B828
0x00007FF703317056
0x00007FF70332C612
0x00007FF70332C576
0x00007FF703315CC2
0x00007FF703315B60
0x00007FF70337CC1D
0x00007FF70337CACA
0x00007FF70339B4E8
0x00007FFEE3F93034 in BaseThreadInitThunk
0x00007FFEE4651431 in RtlUserThreadStart

```

**Comment by loganran** [comment]
I can try on Tuesday.

**Comment by baklava151** [comment]
Just to be safe and make sure ProcMon logs are grabbed correctly

1. Start ProcMon
2. Run the installer
3. Once the installer is done, open ProcMon and hit Ctrl+E to stop it
4. File -> Save, make sure to hit All Events before saving

This will drop a PML file, attach it here

**Comment by fernandex00** [comment]
i got a custom theme, and ultrauxthemepatcher i think thats why the app cannot run

**Comment by spikespaz** [comment]
@fernandex00 I'm not sure why that would be causing the issue. Why would a theme patcher prevent the program from writing and reading its own registry keys?

**Comment by spikespaz** [comment]
I think this issue may be resolved. Take a look at my comment on another issue [here](https://github.com/spikespaz/search-deflector/issues/8#issuecomment-431235680).

**Comment by spikespaz** [comment]
For all of you having this issue, email me at promo@spikespaz.com, and link your GitHub profile in the body of the message. I will send a one-time use promotion code for the Microsoft Store version ([here](https://www.microsoft.com/store/productId/9P8ZJJ80RZ2K)) to compensate for your troubles. Additionally, you won't have this issue with the Store version.

**Comment by blackcomb-dev** [comment]
@spikespaz is this issue also fixed as of 0.3.5?

**Comment by spikespaz** [comment]
@blackcomb-dev Was just about to ask that.

**Comment by spikespaz** [comment]
I need an update. Has this issue been fixed as of 0.3.5 or 0.3.6?

**Comment by loganran** [comment]
I can try again at work tomorrow before i reimage my machine and let you
know.

**Comment by spikespaz** [comment]
@loganran @tango409 Have either of you had the chance to test the new version?

**Comment by spikespaz** [comment]
Going to close this and assume the issue is resolved unless someone can respond.
